src/cmd/ksh93/tests/emacs.exp

```diff
@@ -0,0 +1,210 @@
+expect -exact ""
+
+# Take cursor to the end of line with ctrl-e
+send -- "\x5"
+expect -exact "set -o emacs"
+
+# Move cursor backward one character with ctrl-b
+send -- "\x2"
+expect -exact "\x8"
+
+# Move cursor forward one character with ctrl-f
+# First move to the beginning of line
+send -- "\x1" 
+expect -exact ""
+send -- "\x6"
+expect -exact "s"
+
+# Move cursor forward one word with alt-f
+send -- "f"
+expect -exact "et"
+
+#
+#
+## ^X^X ?
+#
+
+# Delete a character with ctrl-d
+# Move to the beginning of line
+send -- "\x1" 
+send -- "\x4"
+expect -exact "et -o emacs"
+
+# Delete current word with alt-d
+send -- "d"
+expect -exact "-o emacs"
+
+## Delete previous word with ALT-backspace
+# Take cursor to the end of line with ctrl-e
+send -- "\x5"
+send -- ""
+expect -exact "-o"
+
+# Clear the commandline with ctrl-c
+send -- "\x3"
+expect -exact ""
+
+# Set command and move to end of line
+send "echo hello"
+send -- "\x5"
+
+# Delete previous word with ALT-h
+send -- "h"
+expect -exact "echo"
+
+# Delete previous word with alt-delete
+send -- "\[3;3~"
+expect -exact ""
+
+
+# Transpose character with ctrl-t
+send -- "echo hello\x14"
+expect -exact "echo helol"
+
+## Capitalize current word with alt-c
+# Clear command line
+send -- "\x3"
+send -- "echo hello"
+# Move to beginning of line
+send -- "\x1"
+# This should capitalize the word 'echo' to 'ECHO'
+send -- "c"
+expect -exact "ECHO"
+
+# Change current word to lowercase alt-l
+# Move to beginning of line
+send -- "\x1"
+send -- "l"
+expect -exact "echo"
+
+# Delete from cursor to end of line with ctrl-k
+send -- "\x1"
+send -- "\x11"
+# Cursor is at beginning of line so that should delete full line
+expect -exact ""
+
+## Kill from cursor to mark with ctrl-w
+send -- "echo hello2"
+send -- "\x17"
+expect -exact ""
+
+#
+## M-p       Push the region from the cursor to the mark on the stack.
+#
+## ^Y        Restore last item removed from line. (Yank item back to the line.)
+send -- "\x19"
+expect -exact "echo hello2"
+
+#
+## ^L        Line feed and print current line
+#send -- ""
+#
+## M-^L      Clear the screen
+#send -- ""
+#
+
+## ^J        (New line) Execute the current line.
+#
+## ^M        (Return) Execute the current line.
+send -- "m"
+expect "\r\nhello2\r\n"
+
+#
+## M-<       Fetch the least recent (oldest) history line.
+send -- "<"
+expect "echo hello"
+#
+## M->       Fetch the most recent (youngest) history line.
+# send -- ">"
+# expect -re "\r\$ echo hello2"
+
+#
+## ^Rstring  Reverse  search  history  for  a previous command line containing string.
+send -- "\x3"
+send -- "echo hello\r"
+expect -re ".*echo hello"
+
+#
+## ^O        Operate - Execute the current line and fetch the next line relative to current line from the history file.
+send -- "echo foobar\xF"
+expect -re ".*foobar"
+#
+# M-.       The last word of the previous command is inserted on the line.
+send -- "."
+expect "foobar"
+
+#
+## M-_       Same as M-.
+send -- "\x3"
```

Comment:
Your char code is incorrect. If you meant an [esc] char that should be `"\x1B"` or `"\033"`.

Reply:
And, of course, even though your current formulation gets the hex string for [esc] wrong it should not result in a SIGABRT from the ksh being tested. So please open an issue showing how to induce this failure mode.